src/map/addPopUp.js

```diff
@@ -3,174 +3,181 @@ import { getCurrentState } from '../store/actions/actions';
-                  if (contentArr.length > 1) {
-                    rowItem[val] = contentArr.join(', ');
-                  }
-                });
-                content =
-                  `<div>` +
-                  `<div><b>${row[layer.popup.header]}</b></div>` +
-                  `<div><center>${Mustache.render(
-                    layer.popup.body,
-                    commaFormatting(layer, rowItem, true)
-                  )}</center></div>` +
-                  `</div>`;
-              } else {
-                content = Mustache.render(layer.popup.body, commaFormatting(layer, rowItem, true));
-              }
-              break;
-            }
-          }
-        } else {
-          // if no data, use feature.properties to populate popup
-          // eslint-disable-next-line no-lonely-if
-          if (feature.properties[layer.popup.header]) {
-            content = `<div><b>${feature.properties[layer.popup.header]}</b></div>`;
-            if (layer.popup.body) {
-              content += `<div><center>${Mustache.render(
-                layer.popup.body,
-                commaFormatting(layer, feature.properties, true)
-              )}</center></div>`;
+            if (layer && layer.type !== 'chart') {
+                // define data to loop through looking for join matches
+                data = (layer.aggregate && layer.aggregate.timeseries) &&
+                    timeseries && timeseries[layerId] && timeseries[layerId].data ? [...timeseries[layerId].data] :
+                    (layer.source && layer.source.data &&
+                        (layer.source.data.features || [...layer.source.data]));
+                if (data && data.length) {
+                    let row;
+                    for (let r = 0; r < data.length; r += 1) {
+                        row = {
+                            ...(data[r].properties || data[r]),
+                        };
+                        const rowItem = {
+                            ...row,
+                            ...feature.properties,
+                        };
+                        // if row matches property
+                        if ((layer.popup.join &&
+                            (row[layer.popup.join[0]] === feature.properties[layer.popup.join[1]])) ||
+                            (!layer.popup.join &&
+                                (row[layer.source.join[1]] === feature.properties[layer.source.join[0]]))) {
+                            const datum = {
+                                ...rowItem,
+                            };
+                            if (layer.popup && layer.popup.hideNulls) {
+                                Object.keys(datum).forEach((k) => {
+                                    if (Number.isNaN(Number(datum[k])) &&
+                                        (datum[k].toLowerCase() === 'n/a' ||
+                                            datum[k] === '')) {
```

Comment:
It feels like we don't need to check if `datum[k]` is a number if we are also going to check if it's these strings?

Reply:
Also, should we add to this conditional so it actually checks for `null`?

Reply:
@cKellyDesign I think n/a and ' ' are most common in data sets we handle. We surely can include null on the list. In terms of checking if it's not a number that's a good call we really don't need to check for that :fist_oncoming: